app/src/androidTest/java/com/github/sdpcoachme/profile/ProfileActivityTest.kt

```diff
@@ -303,6 +293,49 @@ class ProfileActivityTest {
         }
     }
 
+    @Test
+    fun editSports() {
+        getDatabase().setCurrentEmail(NON_COACH_2.email)
+        ActivityScenario.launch<ProfileActivity>(defaultIntent).use {
+            waitForUpdate(it)
+
+            composeTestRule.onNodeWithTag(SPORTS, useUnmergedTree = true).performClick()
+
+            for (rowTag in ROW_TEXT_LIST) {
+                composeTestRule.onNodeWithTag(rowTag.ROW, useUnmergedTree = true).performClick()
+            }
+            composeTestRule.onNodeWithTag(SelectSportsActivity.TestTags.Buttons.DONE, useUnmergedTree = true).performClick()
+
+            waitForUpdate(it)
+            // Given that we click on all sports, the list of sports is the complement
+            for (sport in Sports.values().toSet() - NON_COACH_2.sports.toSet()) {
```

Comment:
As I understand it, Sports.values() contains all possible sports, but NON_COACH_2.sports at this point also contains all sports since ticked everything at line 305. Therefore, the for loop will do nothing. Am I wrong at this?

Reply:
UserInfo is an immutable data class, meaning it will be replaced in the database with a new instance with updated sports. That means that NON_COACH_2 still contains the old sports value, which is why we need to do the complement. That way, we verify that the UI effectively reflects the sports selected in the activity, (click on all sports -> the resulting sports are the complement of the initially selected sports). Hope that makes sense, don't hesitate if you need more details.

Reply:
Okay, makes sense:)